File: app/src/main/java/com/itmoldova/list/ArticlesContract.java
package com.itmoldova.list;

import com.itmoldova.model.Category;
import com.itmoldova.model.Item;

import java.util.List;

/**
 * Author vgrec, on 09.07.16.
 */
public interface ArticlesContract {

    interface View {
        void showArticles(List<Item> items, boolean clearDataSet);

        void setLoadingIndicator(boolean loading);

        void showError();

        void showNoInternetConnection();

        void openArticleDetail(List<Item> items, Item item);

    }

    interface Presenter {
        void loadArticles(Category category, int page);

        void refreshArticles(Category category);

        void cancel();

        void onArticleClicked(List<Item> items, Item item);
    }

}

File: app/src/main/java/com/itmoldova/list/ArticlesContract.kt
package com.itmoldova.list

import android.widget.ImageView
import com.itmoldova.model.Category
import com.itmoldova.model.Article

/**
 * Author vgrec, on 09.07.16.
 */
interface ArticlesContract {

    interface View {
        fun showArticles(articles: List<Article>, clearDataSet: Boolean)

        fun setLoadingIndicator(loading: Boolean)

        fun showError()

        fun showNoInternetConnection()

        fun openArticleDetail(articles: List<Article>, article: Article, imageView: ImageView)

    }

    interface Presenter {
        fun loadArticles(category: Category, page: Int)

        fun refreshArticles(category: Category)

        fun cancel()

        fun onArticleClicked(articles: List<Article>, article: Article, imageView: ImageView)
    }

}
